physics.py

```diff
@@ -171,12 +178,96 @@ def update(self, tm_diff: float, robotVoltage: float) -> None:
             [chassisSpeed.vx, chassisSpeed.vy, chassisSpeed.omega],
         )
 
-        deltaTrans = Transform2d(deltaX, deltaY, deltaHeading)
+        deltaTrans = Transform2d(deltaX, -deltaY, deltaHeading)
 
         newPose = self.pose + deltaTrans
         self.pose = newPose
 
 
+class NoteSim:
+    def __init__(self) -> None:
+        self.midlineNotes = constants.kNotesStartingMidline
+        self.blueNotes = constants.kNotesStartingBlueWing
+        self.redNotes = constants.kNotesStartingRedWing
+
+        self.loadingNotes = [
+            constants.kNoteLoadingStationPositionBlue,
+            constants.kNoteLoadingStationPositionRed,
+        ]
+
+    def canPickup(self, note: Pose3d, botPose) -> bool:
+        if pointInCircle(botPose.translation(), note.toPose2d().translation(), 0.5):
```

Comment:
I am confused.

Reply:
if the note is within 0.5m of the center of the bot, consider a circle surrounding the bot of radius 0.5m, the point the note is located at must be inside to be considered "pickupable"